flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaCommittableSerializer.java

```diff
@@ -46,6 +46,15 @@ public byte[] serialize(KafkaCommittable state) throws IOException {
 
     @Override
     public KafkaCommittable deserialize(int version, byte[] serialized) throws IOException {
+        switch (version) {
+            case 1:
```

Comment:
How do we know if there wasn't historical versions where the version number is `0`?

Is there existing migration IT tests that verify, with this change, we can still safely restore from previous versions?

In general, it looks like we're missing some documentation where we historically track what versions have existed before and what their corresponding schema is. I think having a class-level Javadoc to document this is already enough.

@chucheng92 do you think you'd be able to do this as part of this PR contribution? Basically, look at historical changes to confirm that this was indeed the only used version number + document its schema in the class-level Javadoc.

Reply:
@tzulitai hi, Gordon. thanks for your reviewing! the version from 1 can be found at https://github.com/apache/flink/pull/16676/files#diff-decc6e90291b6f7c78efae7bd3223d218a1aba00bbbf53e4c696d63506e907cf. however, we can't always check PR to see if there is an old version or a version change. So I agree with you to add JavaDocs to record serialized version evolutions. I added it in the pr.  WDYT?